tools/additional_lint

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+import sys
+from collections.abc import Callable, Iterable
+from dataclasses import dataclass
+from pathlib import Path
+
+from shared.common import DATA_DIR, REPO_DIR, SRC_DIR, TOOLS_DIR
+
+EXTENSIONS = [
+    "Dockerfile",
+    ".h",
+    ".c",
+    ".build",
+    ".ninja",
+    ".py",
+    ".cs",
+    ".xaml",
+    ".csproj",
+    ".txt",
+    ".rc",
+    ".glsl",
+    ".md",
+    ".json",
+    ".json5",
+    "Dockerfile",
+]
+
+
+@dataclass
+class LintWarning:
+    path: Path
+    message: str
+    line: int | None = None
+
+    def __str__(self) -> str:
+        prefix = str(self.path.relative_to(REPO_DIR))
+        if self.line is not None:
+            prefix += f":{self.line}"
+        return f"{prefix}: {self.message}"
+
+
+def get_files() -> Iterable[Path]:
```

Comment:
Is it possible to exclude the following directories here?

`tools/config/TR1X_ConfigTool/obj`
`tools/config/TR1X_ConfigTool/bin`

These will contain auto-generated files that trigger warnings e.g.

```
tools/config/TR1X_ConfigTool/obj/project.assets.json: missing newline character at end of file
tools/config/TR1X_ConfigTool/obj/TR1X_ConfigTool.csproj.nuget.dgspec.json: missing newline character at end of file
tools/config/TR1X_ConfigTool/obj/Release/net6.0-windows/win-x64/App.g.cs:36: trailing whitespace
...
tools/config/TR1X_ConfigTool/bin/Release/net6.0-windows/win-x64/TR1X_ConfigTool.deps.json: missing newline character at end of file
tools/config/TR1X_ConfigTool/bin/Release/net6.0-windows/win-x64/TR1X_ConfigTool.runtimeconfig.json: missing newline character at end of file
```

Reply:
Done, please LMK if it works as intended now.

Reply:
Works perfectly 👍